pom.xml

```diff
@@ -0,0 +1,339 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<groupId>org.vassal</groupId>
+	<artifactId>vassalengine</artifactId>
+	<version>3.3.1-SNAPSHOT</version>
+
+	<properties>
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+		<project.reporting.outputEncoding>${project.build.sourceEncoding}</project.reporting.outputEncoding>
+		<version.java.build>11</version.java.build>
+		<version.java.target>11</version.java.target>
+
+		<vassal.jarlocation>${project.build.directory}/lib</vassal.jarlocation>
+
+		<version.batik>1.13</version.batik>
+		<version.jmock>2.5.1</version.jmock>
+	</properties>
+
+	<name>Vassal Engine</name>
+	<inceptionYear>2001</inceptionYear>
+	<url>https://github.com/vassalengine/vassal/</url>
+
+	<dependencies>
+		<dependency>
+			<groupId>ch.qos.logback</groupId>
+			<version>1.2.3</version>
+			<artifactId>logback-classic</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>com.miglayout</groupId>
+			<artifactId>miglayout-swing</artifactId>
+			<version>4.2</version>
+		</dependency>
+		<dependency>
+			<groupId>javazoom</groupId>
+			<artifactId>jlayer</artifactId>
+			<version>1.0.1</version>
+		</dependency>
+		<dependency>
+			<groupId>jivesoftware</groupId>
+			<artifactId>smack</artifactId>
+			<version>3.0.4</version>
+		</dependency>
+		<dependency>
+			<groupId>jivesoftware</groupId>
+			<artifactId>smackx</artifactId>
+			<version>3.0.4</version>
+		</dependency>
+		<dependency>
+			<groupId>commons-codec</groupId>
+			<artifactId>commons-codec</artifactId>
+			<version>1.13</version>
+		</dependency>
+		<dependency>
+			<groupId>commons-io</groupId>
+			<artifactId>commons-io</artifactId>
+			<version>2.6</version>
+		</dependency>
+		<dependency>
+			<groupId>gnu.getopt</groupId>
+			<artifactId>java-getopt</artifactId>
+			<version>1.0.13</version>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.commons</groupId>
+			<artifactId>commons-lang3</artifactId>
+			<version>3.9</version>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.xmlgraphics</groupId>
+			<artifactId>batik-codec</artifactId>
+			<version>${version.batik}</version>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.xmlgraphics</groupId>
+			<artifactId>batik-dom</artifactId>
+			<version>${version.batik}</version>
+			<exclusions>
+				<exclusion>
+					<groupId>xalan</groupId>
+					<artifactId>xalan</artifactId>
+				</exclusion>
```

Comment:
Looks like we need to add this:
```
<exclusion>
   <groupId>xml-apis</groupId>
   <artifactId>xml-apis</artifactId>
</exclusion>
```

Reply:
I added that to my pom.xml and that fixed all of the XML relate errors, except now I am back to the original problem of

import sun.java2d.cmm.ProfileDeferralMgr;

not being resolved.

Reply:
I have just spent a few hours trying to get this to run on Eclipse. Found the xml-apis issue as well and this same fix for it. But I also ran into another much bigger problem which has to do with Eclipse, Maven and JPMS (Java Platform Module System).

We use compiler arguments --add-exports, only one of them seems to be really needed for the ProfileDeferralMgr. But due to unfixed https://bugs.eclipse.org/bugs/show_bug.cgi?id=543631 the Eclipse Maven plugin (M2E) is not able to pass compiler arguments to the Eclipse project configuration. These arguments can be set manually in Eclipse, but they will be reset after each refresh by M2E.

Possible solutions that I see, in order of preference:
1. Check if this `ProfileDeferralMgr.deferred = false` workaround is still needed, there were some updates to ImageIO in the last years, with some luck this workaround and all the --add-exports arguments can be removed
2. Tell Eclipse users to switch to IntelliJ, the compiler argument has to be added there but adding them once is enough and they do not get overwritten by Maven
3. Tell Eclipse users to manually add the --add-exports after every update of the project configuration by M2E (not a good solution)
4. Do not switch to Maven, write an instruction for Eclipse and IntelliJ on how to set the compiler argument once (even worse than 3.)

Reply:
I wonder why that works for me from the command line. Hmm.

Reply:
What exactly is the error message?

Reply:
The import sun.java2d.cmm.ProfileDeferralMgr cannot be resolved.

Reply:
Also, I have no great objection to moving to Intellij. I only use Eclipse because it is what I started using many years ago.

Reply:
I happen to have the Java 13 source handy; `ProfileDeferralMgr` is still just as non-thread safe as it was when I first added the workaround. If we removed the workaround, we'd start getting  `ConcurrentModificationException` reports again as soon as we did.

There might be a different way of dealing with the problem if importing `ProfileDeferralMgr` is such a stumbling block now. I'll have a look at that tomorrow.

Aside: This is one of the most aggravating bugs I've encountered in the JDK. It throws an unchecked exception that violates every expectation of how image loaders should work. It was reported in 2010, and the fix would take 30 seconds to apply---all that's needed is a `synchronized` block inside `ProfileDeferralMgr.activateProfiles()`.

Reply:
It works from the command line all right, the problem is getting the Eclipse IDE to show the code as compileable. Eclipse uses its own compile settings, these can be set manually when maven is not in use, this is how these compiler flags need to be set in Eclipse: http://www.vassalengine.org/forum/viewtopic.php?f=5&t=11991#p61234.

But, when maven is in use, the build configuration of Eclipse is set by maven. The problem is that maven does not pass the compiler arguments, the `--add-export`s in particular, into Eclipse's build configuration. Instead it resets these manual export settings to no exports. This leads to Eclipse not knowing how to compile ImageIOImageLoader and showing these two errors for that class:
```
The import sun.java2d.cmm cannot be resolved
ProfileDeferralMgr cannot be resolved to a variable
```

I just tried it again and it reproducible, I can import the maven project, Eclipse shows these two errors, then I manually add the --add-exports in this very complicated dialog, then the errors are gone. Then I do a "Update Maven Project" (alt+f5), and my manual export settings are gone and the two errors are back.

Reply:
It seems 3 out of 4 compiler args were not needed anymore, it is only the `sun.java2d.cmm` one that is still needed for the `ProfileDeferralMgr.deferring = false` workaround.

Reply:
My updated solutions to the `--add-exports` Eclipse problem, in order of preference:
1. Check if this `ProfileDeferralMgr.deferring = false` workaround is still needed, there were some updates to ImageIO in the last years, with some luck this workaround and the `--add-exports` compiler argument can be removed
2. Declare the Eclipse Java IDE as not supported for Vassal development. Suggest to use another IDE or editor. Write a step by step guide on how to get started with another code editor. I can help with IntelliJ.
3. Find a library for rendering images and rewrite the rendering code to use that library instead of ImageIO, thereby removing the need for the workaround and the `--add-exports` compiler arg
4. Tell Eclipse users to manually add the --add-exports after every update of the project configuration by M2E (not a good solution)
5. Do not switch to Maven, write an instruction for Eclipse and IntelliJ on how to set the compiler argument once (even worse than solution 4.)

Reply:
Just now read your other comments here, so removing the workaround seems to be a non solution.

Can we drop support for Eclipse then? @BrentEaston is willing to try something else, @uckelman appears to not use an IDE at all and probably uses some command-line tool like `ed` or a brain-computer-interface that changes the bits on the disk drive, @Cattlesquat already got a bit jealous of the ease of use of IntelliJ.